pkg/component/watch.go

```diff
@@ -0,0 +1,204 @@
+package component
```

Comment:
was just wondering why are we creating a new file, we can add the functions to component/component.go like all the other functions, right?

Reply:
We can, I just thought that it this is a nice piece that can be put in the separate file, so we don't end up with everything in one huge file.

I actually don't mind putting it into `component.go` if you think that it is better to have it there.

Reply:
yeah, I'm okay :man_shrugging: